bundle/edu.gemini.ui.miglayout/src/main/scala/edu/gemini/ui/miglayout/MigPanel.scala

```diff
@@ -0,0 +1,337 @@
+  case class NoUnit(value: Int) extends MigUnits[Int] {
+    override def toBoundSize = value.toString
+  }
+
+  case class PixelsUnit(value: Int) extends MigUnits[Int] {
+    override def toBoundSize = s"${value}px"
+  }
+
+  case class PercentUnit[T: Numeric](value: T) extends MigUnits[T] {
+    override def toBoundSize = s"$value%"
+  }
+
+  // Converts Ints and Doubles to Units
+  implicit class Int2Unit(val value: Int) extends AnyVal {
+    def px:MigUnits[Int] = PixelsUnit(value)
+    def pct:MigUnits[Int] = PercentUnit(value)
+  }
+
+  implicit class Double2Unit(val value: Double) extends AnyVal {
+    def pct:MigUnits[Double] = PercentUnit(value)
+  }
+
+  /**
+   * Decorator for the LC class so we can construct as LC()
+   * Note that MigCC is final otherwise we'd rather extend it
+   */
+  object LC {
+    def apply(): MigLC = new MigLC()
+  }
+
+  // Add operations to LC()
+  implicit class LCOps(val lc: MigLC) extends AnyVal {
+    /**
+     * Create insets all around the same value in pixels
+     */
+    def insets(s: Int):MigLC =
+      lc.insets(s"${s}px", s"${s}px", s"${s}px", s"${s}px")
+
+    /**
+     * Create insets all around the same value
+     */
+    def insets[T](s: MigUnits[T]):MigLC =
+      lc.insets(s.toBoundSize, s.toBoundSize, s.toBoundSize, s.toBoundSize)
+
+    /**
+     * Create insets in pixels
+     */
+    def insets(top: Int, left: Int, bottom: Int, right: Int):MigLC =
+      lc.insets(s"${top}px", s"${left}px", s"${bottom}px", s"${right}px")
+
+    /**
+     * Create insets
+     */
+    def insets[T, U, V, W](top: MigUnits[T], left: MigUnits[U], bottom: MigUnits[V], right: MigUnits[W]):MigLC =
+      lc.insets(top.toBoundSize, left.toBoundSize, bottom.toBoundSize, right.toBoundSize)
+
+    /**
+     * insetsAll in units
+     */
+    def insetsAll[T](s: MigUnits[T]):MigLC =
+      lc.insetsAll(s.toBoundSize)
+
+    /**
+     * Set alignment on X and Y in one call
+     */
+    def align(x: HMigAlign, y: VMigAlign): MigLC = lc.align(x.toAlign, y.toAlign)
+
+    /**
+     * Set grid X gaps in Units
+     */
+    def gridGapX[T](g: MigUnits[T]) = lc.gridGapX(g.toBoundSize)
+
+    /**
+     * Set grid Y gaps in Units
+     */
+    def gridGapY[T](g: MigUnits[T]) = lc.gridGapY(g.toBoundSize)
+
+    /**
+     * Set grid gap on X and Y in Units
+     */
+    def gridGap[T, U](x: MigUnits[T], y: MigUnits[U]) = lc.gridGap(x.toBoundSize, y.toBoundSize)
+  }
+
+  /**
+   * Decorator for the MigAC class so we can construct as CC()
+   * Note that MigCC is final otherwise we'd rather extend it
+   */
+  object AC {
+    def apply(): MigAC = new MigAC()
+  }
+
+  /**
+   * Decorator for the MigCC class so we can construct as CC()
+   * Note that MigCC is final otherwise we'd rather extend it
+   */
+  object CC {
+    def apply(): MigCC = new MigCC()
+  }
+
+  // Structural type for a class that support alignment
+  private type Alignable[T] = {
+    def alignX(s: String): T
+    def alignY(s: String): T
+  }
+
```

Comment:
Not a big deal since this is internal to the implementation, but It's generally better to use a typeclass in situations like this.